resolwe_bio/tools/make_amplicon_table.py

```diff
@@ -0,0 +1,172 @@
+            common_columns = [x for x in pick_columns if x in header]
+            if pick_columns:
+                # Find indexes of selected columns
+                temp_header = {col: i for i, col in enumerate(header)}
+                pick_indexes = [temp_header[col] for col in common_columns]
+                header = common_columns
+        for line in tfile:
+            line_content = line.strip().split(delimiter)
+            if pick_columns:
+                # Select only specific columns and order them as in pick_columns:
+                temp_line_content = {i: col for i, col in enumerate(line_content)}
+                line_content = [temp_line_content[i] for i in pick_indexes]
+            table_data.append(line_content)
+    return table_data, header
+
+
+def snp_url(snpid):
+    """Create url SNP ID."""
+    if snpid.startswith('rs'):
+        url = 'http://www.ncbi.nlm.nih.gov/snp/?term={}'.format(snpid)
+        pass
+    elif snpid.startswith('COSM'):
+        url = 'http://grch37-cancer.sanger.ac.uk/cosmic/mutation/overview?id={}'.format(snpid.lstrip('COSM'))
+    elif snpid.startswith('COSN'):
+        url = 'http://cancer.sanger.ac.uk/cosmic/ncv/overview?id={}'.format(snpid.lstrip('COSN'))
+    else:
+        return snpid
+    return url
+
+
+def main():
+    """Invoke when run directly as a program."""
+    args = parse_arguments()
+
+    amplicons = {}
+    amp_cov = {}
+    variants = []
+
+    with open(args.master_file) as master_file:
+        for line in master_file:
+            amplicon = line.strip().split('\t')
+
+            amplicons[amplicon[3]] = HTSeq.GenomicInterval(
+                amplicon[0],
+                int(amplicon[1]),
+                int(amplicon[2])
+            )
+
+    with open(args.amplicon_coverage) as ac:
+        for line in ac:
+            cov = line.strip().split('\t')
+            amp_cov[cov[0]] = float(cov[1])
+
+    for annot_var in args.variants:
+        columns = ['CHROM', 'POS', 'REF', 'ALT', 'AF', 'DP', 'DP4', 'GEN[0].AD', 'SB', 'FS', 'EFF[*].GENE', 'ID']
+        data, header = _tsv_to_list(annot_var, has_header=True, pick_columns=columns)
+
+        for var in data:
+            variant = dict(zip(header, var))
+            gb_pos = '{}:{}'.format(variant['CHROM'], int(variant['POS']))
+            var_or_indel = '{}>{}'.format(variant['REF'], variant['ALT'])
+
+            variant_pos = HTSeq.GenomicPosition(variant['CHROM'], int(variant['POS']))
+            for amp in amplicons:
+                if variant_pos.is_contained_in(amplicons[amp]):
+                    cov = amp_cov[amp]
+                    break
+
+            if 'DP4' in variant:
+                process = 'LoFreq'
+                other_fields = 'DP4={};SB={}'.format(variant['DP4'], variant['SB'])
+            else:
+                process = 'GATK'
+                other_fields = 'AD:{};FS:{}'.format(variant['GEN[0].AD'], variant['FS'])
+
+            if '' in variant['ID'].split(';'):
```

Comment:
In case `variant['ID']` is equal to `'ABC;'`, you can loose some data. It is probably better to check:
`if variant['ID'].strip(';') == '':`